src/psyclone/tests/psyir/nodes/array_mixin_test.py

```diff
@@ -168,3 +252,31 @@ def test_get_effective_shape(fortran_reader, fortran_writer):
     with pytest.raises(NotImplementedError) as err:
         _ = routine.children[4].lhs._get_effective_shape()
     assert "include a function call or expression" in str(err.value)
+
+
+# get_outer_range_index
+
+def test_get_outer_range_index():
+    '''Check that the get_outer_range_index method returns the outermost index
+    of the children list that is a range. Use ArrayReference and
+    ArrayOfStructuresReference as concrete implementations of ArrayMixins.
+    '''
+    symbol = DataSymbol("my_symbol", ArrayType(INTEGER_TYPE, [10, 10, 10]))
+    array = ArrayReference.create(symbol, [Range(), Range(), Range()])
+    assert array.get_outer_range_index() == 2
+
+    symbol = DataSymbol("my_symbol", DeferredType())
+    aos = ArrayOfStructuresReference.create(
+        symbol, [Range(), Range(), Range()], ["nx"])
+    assert aos.get_outer_range_index() == 3  # +1 for the member child
```

Comment:
That doesn't look right/useful to me - I think it would make more sense if it returned the index of the entry in `indices` that was a Range.

Reply:
I'm not sure why I added this test as I did not modify the original implementation. I think I started added all missing tests for this file then gave up (they are implemented in other classes seeing as this is a mixin class, but I don't think that is right). It is used in a couple of transformations at the moment. Any change to its implementation is out of scope for this PR I would strongly argue.